Hardcore.lua

```diff
@@ -212,6 +213,14 @@ end
 SLASH_HARDCORE1, SLASH_HARDCORE2 = '/hardcore', '/hc'
 SlashCmdList["HARDCORE"] = SlashHandler
 
+--[[ Override default WoW UI ]]--
+
+TradeFrameTradeButton:SetScript("OnClick", function()
+	table.insert(Hardcore_Character.trade_partners, TradeFrameRecipientNameText:GetText())
+	Hardcore_Character.trade_partners = Hardcore_FilterUnique(Hardcore_Character.trade_partners)
+	AcceptTrade()
+end)
```

Comment:
After semi successfully trying to use trade events to track trades, I came up with this solution.
Technically this does not track whether a trade finished, but rather whether the player clicked the "Accept" button.
Clicking the Accept button shows clear intent of the player to trade, so I think it is a good enough condition to use.

Reply:
Not sure if this is dangerous, but I'm assuming its okay. Does hooking the script work?

Reply:
Surprisingly yes. I read in the docs that `AcceptTrade` is a protected function, but it works. Wanna hop on real quick to test it? It would be good to test that it works for both trade parties, I only tested it on my end.

Reply:
As for is it dangerous, I guess it could get overwritten by other addons that get loaded after the hardcore addon 😕

Reply:
Oh I think accept trade is protected but not the OnClick. Sweet!